Sources/JWTKeychain/Controllers/API/UserController.swift

```diff
@@ -66,13 +72,109 @@ open class UserController: UserControllerType {
 
     open func regenerate(request: Request) throws -> ResponseRepresentable {
         let user = try request.user()
-        let token = try self.configuration.generateToken(userId: user.id!)
+        let token = try self.configuration.generateToken(user: user)
         return try JSON(node: ["token": token])
     }
 
     open func me(request: Request) throws -> ResponseRepresentable {
         let user = try request.user()
-        let token = try self.configuration.generateToken(userId: user.id!)
+        let token = try self.configuration.generateToken(user: user)
         return try user.makeJSON(token: token)
     }
+
+    open func resetPasswordEmail(request: Request) -> ResponseRepresentable {
+        do {
+
+            guard
+                let email = request.data["email"]?.string,
+                let user: User = try User.query().filter("email", email).first() else {
+                    return JSON(["success": "Instructions were sent to the provided email"])
```

Comment:
Part of it yes. If the email is missing. But due to security reasons we shouldn't tell if the email exists or not on the database and just return the same email on success or failure.